mealsharingdotnet/Program.cs

```diff
@@ -0,0 +1,167 @@
+
+using System.Linq;
+using System.Numerics;
+using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddScoped<IMealService, FileMealService>();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+  app.UseSwagger();
+  app.UseSwaggerUI();
+}
+
+app.MapGet("/", () => "week5-mealsharing dotnet");
+
+/*
+ <summary>
+ Endpoint for adding a new blog post.
+ <example>
+ postman input
+ <code>
+ {
+     "headline": "Nepal Hot Pot",
+         "imageURL": "https://askme.com",
+         "bodyText": "popular dumpling",
+         "location": "ishoj,cph",
+         "price": 500
+ }
+ </code>
+ </example>
+ </summary>
+*/
+app.MapPost("/meals", ([FromServices] IMealService mealSharingService, Meal meal) =>
+{
+  mealSharingService.AddMeal(meal);
+});
+
+/*
+ <summary>
+ Endpoint for getting all blog posts.
+ <example>
+ postman input
+ <code>
+ http://localhost:5157/meals
+ </code>
+ </example>
+ </summary>
+*/
+app.MapGet("/meals", ([FromServices] IMealService mealSharingService) =>
+{
+  return mealSharingService.ListMeals();
+});
+
+/*
+ <summary>
+ Endpoint for updating a blog posts by id.
+ <example>
+ postman input
+ <code>
+ http://localhost:5157/meals/1
+ </code>
+ </example>
+ </summary>
+*/
+app.MapPut("/meals/{id}", ([FromServices] IMealService mealSharingService, int id, Meal updateBody) =>
+{
+  mealSharingService.UpdateMeal(id, updateBody);
+});
+
+/*
+ <summary>
+ Endpoint for deletting a blog posts by id.
+ <example>
+ postman input
+ <code>
+ http://localhost:5157/meals/1
+ </code>
+ </example>
+ </summary>
+*/
+app.MapDelete("/meals/{id}", ([FromServices] IMealService mealSharingService, int id) =>
+{
+  mealSharingService.DeleteMeal(id);
+});
+
+app.Run();
+
+public interface IMealService
+{
+  List<Meal> ListMeals();
+  void AddMeal(Meal meal);
+  void DeleteMeal(int id);
+  void UpdateMeal(int id, Meal updateBody);
```

Comment:
All of these methods can use `Task` and `async` (and when you call them and in the logic) 😊